# **Data Collection Notebook**

## Objectives

* Fetch data from Kaggle and save it as raw data.
* Inspect the data and save it under `outputs/datasets/collection`

## Inputs

* Kaggle JSON file — the authentication token.

## Outputs

* Generate Dataset: `outputs/datasets/collection/ai_job_dataset1.csv`

## Additional Comments

* Dataset: **Global AI Job Market & Salary Trends 2025** by Bisma Sajjad on Kaggle.
* 15,000+ synthetic job listings from 50+ countries with 20 features.
* In the workplace, data is not pushed to public repositories for security reasons.


---

# Change working directory

We need to change the working directory from its current folder to its parent folder.
* We access the current directory with `os.getcwd()`

In [6]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\chahi\\Desktop\\vscode-project\\the-ai-salary-index\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory.
* `os.path.dirname()` gets the parent directory
* `os.chdir()` defines the new current directory

In [7]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [8]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\chahi\\Desktop\\vscode-project\\the-ai-salary-index'

# Fetch data from Kaggle

Install the Kaggle package to fetch data.

In [9]:
%pip install kaggle==2.0.0

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.0.1 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


You need a `kaggle.json` file (authentication token) in the project root.
* Download it from your Kaggle account under **Account → API → Create New API Token**.

In [10]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = os.getcwd()

Define the Kaggle dataset path and the destination folder, then download it.

In [11]:
KaggleDatasetPath = "bismasajjad/global-ai-job-market-and-salary-trends-2025"
DestinationFolder = "inputs/datasets/raw"
os.makedirs(DestinationFolder, exist_ok=True)

! kaggle datasets download -d {KaggleDatasetPath} -p {DestinationFolder}

Dataset URL: https://www.kaggle.com/datasets/bismasajjad/global-ai-job-market-and-salary-trends-2025
License(s): CC0-1.0




  0%|          | 0.00/1.08M [00:00<?, ?B/s]
 93%|█████████▎| 1.00M/1.08M [00:00<00:00, 1.64MB/s]
100%|██████████| 1.08M/1.08M [00:00<00:00, 1.66MB/s]


Unzip the downloaded file.

In [12]:
import glob
import zipfile

# Find all zip files in the folder
zip_files = glob.glob(os.path.join(DestinationFolder, "*.zip"))

# Extract each zip file and then delete it
for zip_path in zip_files:
    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
        zip_ref.extractall(DestinationFolder)
    os.remove(zip_path)

# Remove kaggle.json if it exists to avoid exposing credentials
kaggle_json = os.path.join(os.getcwd(), "kaggle.json")
if os.path.exists(kaggle_json):
    os.remove(kaggle_json)

print("All ZIP files extracted and deleted.")

All ZIP files extracted and deleted.


# Inspect the data

In [13]:
import pandas as pd
df = pd.read_csv(f"{DestinationFolder}/ai_job_dataset1.csv")
print(f"Shape: {df.shape}")
df.head()

Shape: (15000, 20)


job_id                 job_title  salary_usd salary_currency  \
0  AI00001            Data Scientist      219728             USD   
1  AI00002                Head of AI      230237             JPY   
2  AI00003             Data Engineer      128890             EUR   
3  AI00004  Computer Vision Engineer       96349             USD   
4  AI00005         Robotics Engineer       63065             EUR   

   salary_local experience_level employment_type company_location  \
0        219728               EX              PT           Sweden   
1      25326070               EX              PT            Japan   
2        109557               EX              CT          Germany   
3         96349               MI              FL          Finland   
4         53605               EN              FT           France   

  company_size employee_residence  remote_ratio  \
0            M             Sweden             0   
1            L              Japan            50   
2            S            Germany           100   
3            L            Finland            50   
4            S             France           100   

                      required_skills education_required  years_experience  \
0  Python, Computer Vision, R, Docker          Associate                13   
1  Kubernetes, MLOps, Tableau, Python           Bachelor                10   
2  Spark, Scala, Hadoop, PyTorch, GCP           Bachelor                12   
3       MLOps, Linux, Tableau, Python                PhD                 2   
4          R, Scala, SQL, GCP, Python          Associate                 0   

         industry posting_date application_deadline  job_description_length  \
0  Transportation   2024-09-23           2024-10-31                    1132   
1  Transportation   2024-07-26           2024-09-12                    2299   
2      Automotive   2025-01-19           2025-03-28                    1329   
3      Automotive   2024-07-20           2024-09-06                    1132   
4         Finance   2025-03-16           2025-05-09                    2011   

   benefits_score           company_name  
0             6.6           TechCorp Inc  
1             8.5     Cloud AI Solutions  
2             5.5  Quantum Computing Inc  
3             6.8    Cognitive Computing  
4             9.3      Advanced Robotics

Check data types and missing values.

##data type:

In [14]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   job_id                  15000 non-null  str    
 1   job_title               15000 non-null  str    
 2   salary_usd              15000 non-null  int64  
 3   salary_currency         15000 non-null  str    
 4   salary_local            15000 non-null  int64  
 5   experience_level        15000 non-null  str    
 6   employment_type         15000 non-null  str    
 7   company_location        15000 non-null  str    
 8   company_size            15000 non-null  str    
 9   employee_residence      15000 non-null  str    
 10  remote_ratio            15000 non-null  int64  
 11  required_skills         15000 non-null  str    
 12  education_required      15000 non-null  str    
 13  years_experience        15000 non-null  int64  
 14  industry                15000 non-null  str    
 

##Missing values

In [15]:
df.isnull().sum()

job_id                    0
job_title                 0
salary_usd                0
salary_currency           0
salary_local              0
experience_level          0
employment_type           0
company_location          0
company_size              0
employee_residence        0
remote_ratio              0
required_skills           0
education_required        0
years_experience          0
industry                  0
posting_date              0
application_deadline      0
job_description_length    0
benefits_score            0
company_name              0
dtype: int64

##Summary statistics

In [16]:
df.describe()

salary_usd  salary_local  remote_ratio  years_experience  \
count   15000.000000  1.500000e+04  15000.000000      15000.000000   
mean   121991.938267  8.292366e+05     50.196667          6.365667   
std     63968.361846  3.425325e+06     40.844084          5.598551   
min     16621.000000  1.662100e+04      0.000000          0.000000   
25%     74978.500000  7.383075e+04      0.000000          2.000000   
50%    107261.500000  1.090355e+05     50.000000          5.000000   
75%    155752.250000  1.673278e+05    100.000000         10.000000   
max    410273.000000  3.368541e+07    100.000000         19.000000   

       job_description_length  benefits_score  
count            15000.000000    15000.000000  
mean              1500.852600        7.499540  
std                574.724647        1.444202  
min                500.000000        5.000000  
25%                998.750000        6.300000  
50%               1512.000000        7.500000  
75%               1994.000000        8.800000  
max               2499.000000       10.000000

**Key observations from summary statistics:**

* **`salary_usd`** — Mean ~$122k, max $410k. The large standard deviation (~$64k) reflects the global spread of salaries across countries and roles. This is the primary target variable for analysis.
* **`salary_local`** — Very high std and max values. This is expected since it mixes different currencies (USD, JPY, EUR, etc.) making direct comparisons unreliable. Not suitable for modelling as-is.
* **`remote_ratio`** — Takes only 3 values: 0 (on-site), 50 (hybrid), 100 (fully remote). The mean of ~50 suggests a fairly balanced distribution across the three categories.
* **`years_experience`** — Ranges from 0 to 19, mean ~6.4. No negative or unrealistic values — the column looks clean.
* **`job_description_length`** — Ranges from 500 to 2499, roughly uniform. Unlikely to be a useful predictor.
* **`benefits_score`** — Ranges from 5 to 10, mean 7.5. No zeros or invalid entries.

**Overall:** No obvious data quality issues (no negative salaries, no placeholder values like 999). The suspiciously uniform distributions across columns are consistent with the dataset being synthetic.

---

# Push files to Repo

* If you do not need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [18]:
import os

try:
  os.makedirs(name='outputs/datasets/collection') # create outputs/datasets/collection folder
except Exception as e:
  print(e)

df.to_csv(f"outputs/datasets/collection/ai_job_dataset1.csv",index=False)

# Conclusions and Next Steps

* The dataset has been successfully downloaded from Kaggle and saved.
* Next step: **02 - JobMarketStudy** — Exploratory Data Analysis and correlation study.